Users/abehenderson/CSX/Callbacks/groupBy.js

```diff
@@ -0,0 +1,65 @@
+// Attempt #2 
+  // Per Cole's Change Request:
+    // Initialize a variable equal to invoking the callback on 'el'
+    // Eliminate 'cache' and use '[el]' to temporarily store values
+
+function groupBy(arrays, callback) {
+  const obj = {};
+  arrays.forEach(el => {
+    let evRes = callback(el); // Cole: what is a better name that 'evRes'?
+    if (!obj[evRes]) {
+      obj[evRes] = [el];
+    } else {
+      [el] = obj[evRes]; // I didn't know I could use '[el]' as a variable like this
```

Comment:
you've gotten into some advanced shit here by accident!
there is a much simpler way. 
on line 11, you've assigned a value to the `evRes` key on the `obj` object.  the value is the element `el` inside an array. 
the directions want you to add further elements to THAT ARRAY if they meet the correct criterium. you don't need line 13 at all; you don't know it but all you're doing there is what's called "destructuring" the value of `obj[evRes]` and assigning the `obj[evRes][0]` to variable `el`. the reason your code works with line 13 included is that it has no effect on line 14.

Reply:
Yes, I see that now.  

I was stuck in the mindset that I needed with the previous way I solved the problem.  I was using `cache` as temporary storage that I would fill and dump with each iteration.  That idea got stuck in my head and carried over, even when you said I didn't need `cache`

Reply:
But here I can see that once `obj[evRes] = [el];` is declared, I can just continue to push to the array `[el]` without the need for temporary storage....am I understanding that correctly?